File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/PokemonBaseStatBoosterModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.Stat;
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class PokemonBaseStatBoosterModifierItem extends PokemonHeldItemModifierItem implements GeneratedPersistentModifierType {

    public static final String TARGET = "PokemonBaseStatBoosterModifierType";

    private Stat stat;
    private String localeName;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/PokemonBaseStatBoosterModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.Stat
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType

class PokemonBaseStatBoosterModifierItem : PokemonHeldItemModifierItem(), GeneratedPersistentModifierType {

    companion object {
        const val TARGET = "PokemonBaseStatBoosterModifierType"
    }

    var stat: Stat? = null
    var localeName: String? = null
}